### PR TITLE
psh: tools: handle empty string and lists in assert_cmd expected argument and other small corrections for psh test module

### PR DESCRIPTION
**Pull request by damianloew**
## Description

Small changes in psh test module (does not affect existing tests):
- fix assert_exec function in psh module (the function hasn't been used before)
- handle passing empty string and tuples in `expected` argument in assert_cmd function
- small corections in README.md for psh test module
- pass pexpect.EOF to `pexpect.expect` in order to print assertion message when EOF occured
## Motivation and Context

JIRA: PD-77

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## How Has This Been Tested?

- [ ] Already covered by automatic testing.
- [ ] New test added: (add PR link here).
- [x] Tested by hand on: (ia32-generic).

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing linter checks and tests passed.
- [x] My changes generate no new compilation warnings for any of the targets.

## Special treatment

- [ ] This PR needs additional PRs to work (list the PRs, preferably in merge-order).
- [ ] I will merge this PR by myself when appropriate.

### CODE REVIEW

**Comment by niewim19** [comment]
Was that change checked against current tests that use it that are either on PR or drafts?

**Comment by damianloew** [comment]
> Was that change checked against current tests that use it that are either on PR or drafts?

Yes, all tests that use changed functions works properly with this PR

File: psh/tools/psh.py [hunk 1]
```diff
@@ -37,16 +37,16 @@ def _readable(exp_regex):
...
     pexpect_proc.sendline(cmd)
     cmd = re.escape(cmd)
+    exp_regex = ''
     if is_regex:
         exp_regex = expected
-    else:
-        exp_regex = ''
-        if not isinstance(expected, tuple):
+    elif expected != '':
+        if not isinstance(expected, tuple) and not isinstance(expected, list):
```
Comment (niewim19) [inline psh/tools/psh.py:49]:
```suggestion
        if not (isinstance(expected, tuple) or isinstance(expected, list)):
```

**Review by niewim19 — CHANGES_REQUESTED** [review]
One small nitpick and we can merge it

**Comment by damianloew** [comment]
> One small nitpick and we can merge it

That's right, looks better. Thanks

**Review by niewim19 — APPROVED** [review]
LGTM

**Pull request merged**